pallets/xcm-core-buyer/src/lib.rs

```diff
@@ -276,33 +376,28 @@ pub mod pallet {
         // tanssi collators. So we cannot force them to provide a complex proof, e.g. against relay
         // state.
         #[pallet::call_index(0)]
-        // TODO: weight
-        #[pallet::weight(T::WeightInfo::force_buy_core())]
+        #[pallet::weight(T::WeightInfo::buy_core())]
         pub fn buy_core(
             origin: OriginFor<T>,
             para_id: ParaId,
-            // since signature verification is done in `validate_unsigned`
-            // we can skip doing it here again.
-            proof: BuyCoreCollatorProof<T>,
+            collator_account_id: T::AccountId,
+            proof: BuyCoreCollatorProof<T::CollatorPublicKey>,
         ) -> DispatchResult {
-            // Signature verification is done in `validate_unsigned`.
-            // We use `ensure_none` here because this can only be called by collators, and we do not
-            // want collators to pay fees.
             ensure_none(origin)?;
 
-            let assigned_collators = T::GetAssignedCollators::get_parathread_collators(para_id);
-            if assigned_collators.is_empty() {
-                return Err(Error::<T>::NoAssignedCollators.into());
-            }
-
-            if !assigned_collators.contains(&proof.account) {
+            let is_valid_collator = T::CheckCollatorValidity::is_valid_collator(
```

Comment:
arent we validating this in `validate_unsigned`?

Reply:
yes. Missed this. Will remove.